File: cases/taint/kotlin/expression-negative/ExpressionNegative.kt
package dataflowbench

object ExpressionNegative {
    fun dfb_source(): Int { // DFB-SOURCE: expression-negative-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: expression-negative-sink

    fun run() {
        val value = dfb_source()
        val computed = (value * 3) + 7 // DFB-WITNESS: expression-negative-computed
        dfb_sink(7)
    }
}

File: cases/taint/typescript/expression-negative/ExpressionNegative.ts
function dfb_source(): number { // DFB-SOURCE: expression-negative-input
  return 4;
}

function dfb_sink(value: number): void {} // DFB-SINK: expression-negative-sink

function run(): void {
  const value: number = dfb_source();
  const computed: number = (value * 3) + 7; // DFB-WITNESS: expression-negative-computed
  dfb_sink(7);
}
